services/course-material/src/components/ContentRenderer/moocfi/Map.tsx

```diff
@@ -0,0 +1,224 @@
+import styled from "@emotion/styled"
+import { useQuery } from "@tanstack/react-query"
+import React, { useContext, useEffect } from "react"
+import { useTranslation } from "react-i18next"
+
+import PageContext from "../../../contexts/PageContext"
+import { fetchStudentCountries, postStudentCountry } from "../../../services/backend"
+import SelectField from "../../../shared-module/components/InputFields/SelectField"
+import useToastMutation from "../../../shared-module/hooks/useToastMutation"
+import useUserInfo from "../../../shared-module/hooks/useUserInfo"
+
+import { countryList } from "./../util/Countries"
+import WorldMap from "./worldMap.svg"
+
+const countryClasses = [".fr", ".us", ".fi", ".ng"]
+const formattedClasses = countryClasses.join(",")
+
+interface MapStyleProps {
+  codes: string[]
+}
+
+const Wrapper = styled.div`
+  display: relative;
+  height: auto;
+  background: #ecf0fa;
+  display: grid;
+  padding: 2rem 0 0 2rem;
+  h3 {
+    color: #687eaf;
+    padding-bottom: 0.5rem;
+    border-bottom: 1px solid #dde2ee;
+    font-weight: 500;
+    font-size: 22px;
+  }
+`
+
+const MapStyles = (props: MapStyleProps) => {
+  const formattedClasses = props.codes.join(",")
+  const MAP_STYLES = `
+  ${formattedClasses} {
+      fill: #374461 !important;
+      opacity: 1;
+    }
+  `
+  return MAP_STYLES
+}
+
+// const StyledMap = styled(WorldMap)<string>`
+//   ${formattedClasses} {
+//     fill: #374461 !important;
+//     opacity: 1;
+//   }
+// `
+const StyledMap = styled(WorldMap)`
+  ${MapStyles}
+`
+
+export type RouteElement = Element | SVGLineElement
+
+export interface MapExtraProps {
+  content?: string
+}
+
+export type MapProps = React.HTMLAttributes<HTMLDivElement> & MapExtraProps
+
+const Map: React.FC<React.PropsWithChildren<React.PropsWithChildren<MapProps>>> = () => {
+  let countryCodes: string[] = []
+  const formattedCountryCodes = countryCodes.join(",")
+  const { t } = useTranslation()
+
+  const pageState = useContext(PageContext)
+  const courseId = pageState.pageData?.course_id
+  const courseInstanceId = pageState.instance?.id
+
+  const userInfo = useUserInfo()
+  const userId = userInfo.data?.user_id
+
+  const getCountries = useQuery(
+    [`course-${courseId}-country`],
+    () => {
+      if (courseId) {
```

Comment:
replace the if with https://github.com/rage/secret-project-331/blob/93bb41ddab4e1bb689598673621d62d78f7256e9/services/course-material/src/components/ContentRenderer/moocfi/ExerciseInChapterBlock/ChapterExerciseListGroupedByPage.tsx#L30